pkgs/development/python-modules/py-cpuinfo/default.nix

```diff
@@ -15,6 +17,10 @@ buildPythonPackage rec {
      sha256 = "10qfaibyb2syiwiyv74l7d97vnmlk079qirgnw3ncklqjs0s3gbi";
   };
 
+  # On Darwin sysctl is used to read CPU information.
+  nativeBuildInputs = lib.optionals stdenv.isDarwin [ sysctl ];
```

Comment:
Curious, why do they need sysctl as a buildtime dependency?

Reply:
Tests depend on it.

Reply:
So why not put it into checkInputs?

Reply:
That is odd, since you already added it to propagatedBuildInputs, which are available during the checkPhase.

Reply:
Don't you need `sysctl` in `propagatedBuildInputs` if you want to use `py-cpuinfo` somewhere?

I've tried removing `sysctl` from `nativeBuildInputs` and the tests fail. It works with `checkInputs` so I amended to use that.

Reply:
Uh yeah, I was mixing this up with something else. Having it in propagatedBuildInputs should be sufficient. It should be available during the checkPhase then.

Reply:
I must be holding it wrong, but it doesn't work with just `propagatedBuildInputs` on `aarch64-darwin` for me.

With this:

```
  # On Darwin sysctl is used to read CPU information.
  propagatedBuildInputs = lib.optionals stdenv.isDarwin [ sysctl ];

  preCheck = ''
    type -f sysctl
    exit 1
  '';
```

I see:

```
pytestCheckPhase
Executing pytestCheckPhase
/nix/store/l4ha6r368qyba7klmzgiyhbrxpcyqdc4-stdenv-darwin/setup: line 88: type: sysctl: not found
```

What am I missing?

Reply:
Maybe we need to patch the path to sysctl in the python code.

Reply:
Updated with a somewhat ugly `substituteInPlace`, let me know if you have something better in mind.